### DISCUSSION TITLE
Correct workflow for local development

### DISCUSSION THREAD

**Discussion by neworderofjamie** [neuromorphs/tonic#280] [Q&A]
I check tonic out from git, build with pip and get errors about:

> Versioning for this project requires either an sdist"
      Exception: Versioning for this project requires either an sdist tarball, or access to an upstream git repository. It's also possible that there is a mismatch between the package name in setup.cfg and the argument given to pbr.version.VersionInfo. Project name tonic was given, but was not able to be found.
      
I (eventually) figured out you can fix it by setting the ``PBR_VERSION`` environment variable but that's far from clear from the error message. Perhaps the problem is that git isn't in my path (Windows) but I wonder if this could be handled more gracefully

**Comment by sheiksadique** [comment]
The versioning of this package is done based on the gut history. So it is critical that you use git clone and not download as a zip. Could you please try this and let us know if the problem persists.

**Comment by neworderofjamie** [comment]
I did git clone but, as I say, git isn't in my path (which I think is a pretty normal thing on Windows). Maybe there should be some fallback for local development or at least a more user-friendly error.

**Reply by fabrizio-ottati** [reply]
Don't you know that developing in Windows is illegal?!?!

JK, installing git with pip should solve this problem though.
